blockedit: correctly handle cursor when appending text

chrisbra opened the pull request:
closes #8288

chrisbra committed:
blockedit: correctly handle cursor when appending text

closes #8288

yegappan commented on src/testdir/test_blockedit.vim:
```diff
@@ -28,4 +28,12 @@ func Test_blockinsert_delete()
   bwipe!
 endfunc
 
+func Test_blockappend_eol_cursor()
```
There is a commented out test for this in the Test_visual_block_mode() function
in test_visual.vim (search for BUG).

chrisbra replied:
ah, missed that part of the message.

chrisbra replied:
BTW: I remember always having wondered, why there is the condition for `bd.is_MAX`. I hope this does not cause any problem.s

chrisbra committed:
remove BUG mention from the test_visual

chrisbra committed:
Fix blockedit mode when moving the cursor.

This attempts to fix the issue, that blockedit is wrong, once you start
moving the cursor (as in issue 8288). It may not always work perfectly,
but at least fixes the currently brocken issue.

dpelle commented on src/testdir/test_visual.vim:
```diff
@@ -806,11 +806,8 @@ func Test_visual_block_mode()
-  "call assert_equal(['aaxa', 'bbxb', 'ccxc'], getline(1, '$'))
-  call assert_equal(['aaxa', 'bbba', 'ccca'], getline(1, '$'))
+  call assert_equal(['aaxa', 'bbxb', 'ccxc'], getline(1, '$'))
+  "call assert_equal(['aaxa', 'bbba', 'ccca'], getline(1, '$'))
```
You uncommentted out the first `assert_equal` (which seems good) but you commented out the 2nd one.
Should the 2nd assert really be commented out?  If yes, we probably need  a comment to say why.  It's not clear to me at least but I did not spend much time looking at the change.

chrisbra replied:
Yes, that commented second one is of course wrong and should be removed

chrisbra replied:
ah, Bram had already fixed it.
